scrapy/utils/misc.py

```diff
@@ -113,7 +113,15 @@ def md5sum(file: IO) -> str:
     >>> md5sum(BytesIO(b'file content to hash'))
     '784406af91dd5a54fbb9c84c2236595a'
     """
-    m = hashlib.md5()
+    warnings.warn(
+        (
+            "The scrapy.utils.misc.md5sum function is deprecated, and will be "
+            "removed in a future version of Scrapy."
+        ),
+        ScrapyDeprecationWarning,
+        stacklevel=2,
+    )
+    m = hashlib.md5(usedforsecurity=False)
```

Comment:
Since whether the function is intended for security or not depends on the caller, I thought it best to deprecate this function and reimplement it as private in the media pipelines, which is where it was being used internally, and where we can ensure it is not being used for security purposes.